en/concept/flight_modes.md

```diff
@@ -1,111 +1,215 @@
-# Flight Modes (Developers) 
+# Flight Modes (Developers)
 
-_Flight Modes_ define how the autopilot responds to user input and controls vehicle movement.
+Flight modes are special operational states that control how the autopilot responds to user input and controls vehicle movement.
 They are loosely grouped into _manual_, _assisted_ and _auto_ modes, based on the level/type of control provided by the autopilot.
 The pilot transitions between flight modes using switches on the remote control or with a ground control station.
 
-Not all flight modes are available on all vehicle types, and some modes behave differently on different vehicle types (as described below).
-Finally, some flight modes make sense only under specific pre-flight and in-flight conditions (e.g. GPS lock).
-The system will not allow transitions to those modes until the right conditions are met.
+Flight modes can be implemented in the PX4 flight stack running on the flight controller, or as PX4 ROS 2 modes running on a companion computer.
+From an external perspective (MAVLink/GCS), the origin of a mode is indistinguishable.
 
-The sections below provide an overview of the modes, followed by a [flight mode evaluation diagram](#flight-mode-evaluation-diagram) that shows the conditions under which PX4 will transition into a new mode.
+This topic links to documentation for the supported modes, compares PX4 FC and ROS2 modes, provides implementation hints, and provides an overview of PX4 modes can be used with MAVLink.
 
-::: info
-User-facing flight mode documentation can be found in:
+## Supported Flight Modes
+
+Not all flight modes are available (or makes sense), on all vehicle types, and some modes behave differently on different vehicle types.
+
+Flight mode documentation for the modes running on the flight controller are listed below:
 
 - [Flight Modes (Multicopter)](../flight_modes_mc/index.md)
 - [Flight Modes (Fixed-Wing)](../flight_modes_fw/index.md)
 - [Flight Modes (VTOL)](../flight_modes_vtol/index.md)
 - [Drive Modes (Differential Rover)](../flight_modes_rover/differential.md)
 - [Drive Modes (Ackermann Rover)](../flight_modes_rover/ackermann.md)
+- [Basic Configuration > Flight Modes](../config/flight_mode.md)
+
+::: info
+At time of writing there are no PX4 ROS 2 flight modes that are considered "supported as an integral part of the PX4 open source offering".
+:::
+
+## PX4 FC Modes vs PX4 ROS2 Flight Modes
+
+With some exceptions it is possible to write a mode in either the FC and companion computer.
```

Comment:
Maybe a good time to introduce the Internal vs External concept, and use that troughout the doc

Reply:
Yes, as long as we agree the terminology - https://github.com/PX4/PX4-user_guide/pull/3577#discussion_r1960875336